addon/-private/system/model/internal-model.js

```diff
@@ -119,7 +119,10 @@ export default class InternalModel {
   constructor(modelName, id, store, data) {
     heimdall.increment(new_InternalModel);
     this.id = id;
-    this._internalId = InternalModelReferenceId++;
+
+    // this ensure ordered set can quickly identify this as unique
+    this[Ember.GUID_KEY] = InternalModelReferenceId++ + 'internal-model';
```

Comment:
understood, the secret handshake this is doing is not obvious at all. Im not surprised it was lost.